### PR TITLE
add onesecgroup provider

### PR DESCRIPTION
**Pull request by jnoss**
Work in progress type and provider for onesecgroup, for the Security
Groups new in OpenNebula 4.12

/CC @tuxmea let me know what you think

### CODE REVIEW

**Comment by jnoss** [comment]
Note, the work in progress is in reference to the tests - the type and provider themselves do work, creating, updating, removing onesecgroups in opennebula (Tested in vagrant Centos 7)

File: lib/puppet/provider/onesecgroup/cli.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#
+# License: APLv2
+#
+# Authors:
+# Based upon initial work from Ken Barber
+# Modified by Martin Alfke
+#
+# Copyright
+# initial provider had no copyright
+# Deutsche Post E-POST Development GmbH - 2014, 2015
```
Comment (tuxmea) [inline lib/puppet/provider/onesecgroup/cli.rb]: maybe you want to update the Copyright and author.

File: lib/puppet/provider/onesecgroup/cli.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  end
+
+  # Check if a secgroup exists by scanning the onesecgroup list
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  # Return the full hash of all existing secgroups
+  def self.instances
+    secgroups = Nokogiri::XML(onesecgroup('list', '-x')).root.xpath('/SECURITY_GROUP_POOL/SECURITY_GROUP')
```
Comment (tuxmea) [inline lib/puppet/provider/onesecgroup/cli.rb]: please check whether the following command works and delivers expected results:
puppet resource onesecgroup
This should create a Puppet DSL list of all available OpenNebula security groups.

File: lib/puppet/type/onesecgroup.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#
+# License: APLv2
+#
+# Authors:
+# Based upon initial work from Ken Barber
+# Modified by Martin Alfke
+#
+# Copyright
+# initial provider had no copyright
+# Deutsche Post E-POST Development GmbH - 2014, 2015
```
Comment (tuxmea) [inline lib/puppet/type/onesecgroup.rb]: author and copyright?

**Comment by tuxmea** [comment]
I did not run the tests. Looking good for a first merge. We will happily accept patches afterwards.

File: lib/puppet/provider/onesecgroup/cli.rb (hunk 2 above)
Comment (jnoss) [inline lib/puppet/provider/onesecgroup/cli.rb]: Yes, that does return correctly:

[oneadmin@vagrant-one ~]$ puppet resource onesecgroup --modulepath /vagrant/modules
onesecgroup { 'default':
  ensure      => 'present',
  description => 'The default security group is added to every network. Use it to add default filter rules for your networks. You may remove this security group from any network by updating its properties.',
  rules       => [{'protocol' => 'ALL', 'rule_type' => 'OUTBOUND'}, {'protocol' => 'ALL', 'rule_type' => 'INBOUND'}],
}
onesecgroup { 'security_group_1':
  ensure      => 'present',
  description => 'Description of Security Group 1.',
  rules       => [{'protocol' => 'TCP', 'rule_type' => 'OUTBOUND'}, {'protocol' => 'UDP', 'rule_type' => 'OUTBOUND'}, {'protocol' => 'TCP', 'range' => '22,8000:8010', 'rule_type' => 'INBOUND'}],
}

**Comment by jnoss** [comment]
@tuxmea Comments addressed, this should be ready for merge- Thanks!

**Comment by tuxmea** [comment]
@jnoss thank you. @rwaffen ready to merge.

**Pull request merged**